File: Invoice/server/models/Item.js
import mongoose from "mongoose";

const ItemSchema = new mongoose.Schema({
  companyId: { type: mongoose.Schema.Types.ObjectId, ref: "Company", required: true },
  name: String,
  description: String,
  unitPrice: Number,
  sku: String,
  createdAt: { type: Date, default: Date.now }
});

export default mongoose.model("Item", ItemSchema);

File: Invoice/invoice_app/lib/models/item.dart
class ItemModel {
  final String id;
  final String name;
  final String? description;
  final double? unitPrice;
  final String? sku;

  ItemModel({
    required this.id,
    required this.name,
    this.description,
    this.unitPrice,
    this.sku,
  });

  factory ItemModel.fromJson(Map<String, dynamic> json) {
    return ItemModel(
      id: json["_id"],
      name: json["name"] ?? "",
      description: json["description"],
      unitPrice: json["unitPrice"] != null
          ? double.tryParse(json["unitPrice"].toString())
          : null,
      sku: json["sku"],
    );
  }
}
